crates/router/src/services/api.rs

```diff
@@ -722,21 +722,24 @@ where
 
     let mut event_type = payload.get_api_event_type();
     let tenant_id = if !state.conf.multitenancy.enabled {
-        common_utils::id_type::TenantId::new_unchecked(DEFAULT_TENANT)
+        common_utils::id_type::TenantId::try_from_string(DEFAULT_TENANT.to_owned())
+            .attach_printable("Unable to get default tenant id")
+            .change_context(errors::ApiErrorResponse::InternalServerError.switch())?
     } else {
-        let request_tenant_id = common_utils::id_type::TenantId::wrap(
-            incoming_request_header
-                .get(TENANT_HEADER)
-                .and_then(|value| value.to_str().ok())
-                .ok_or_else(|| errors::ApiErrorResponse::MissingTenantId.switch())?
-                .to_string(),
-        )
-        .change_context(
-            errors::ApiErrorResponse::InvalidRequestData {
-                message: format!("`{}` header is invalid", headers::X_TENANT_ID),
-            }
-            .switch(),
-        )?;
+        let request_tenant_id = incoming_request_header
+            .get(TENANT_HEADER)
+            .and_then(|value| value.to_str().ok())
+            .ok_or_else(|| errors::ApiErrorResponse::MissingTenantId.switch())
```

Comment:
Header parsing error and header missing error shouldn't be the same.

Reply:
Getting parsing here is very unlikely here, we only will be defining the tenant_id in configs. And it will be a valid UTF-8 characterset.